### DISCUSSION TITLE
Unable to convert Oracle trace file to tcl

### DISCUSSION THREAD

**Discussion by chensx712** [TPC-Council/HammerDB#588] [Q&A] [answered]
I am using hammerDB 4.4 and Oracle version 19c. I followed the doc to create a trigger to capture sql trace files. when using hammerDB GUI or convert.tcl to convert the trace file it can not find the binding values. Also I tested with simple trace file it's the same issue. Any idea?

Trace file
---------------
PARSING IN CURSOR #139624246405640 len=212 dep=0 uid=691 oct=3 lid=691 tim=14055447493174 hv=1260924845 ad='fe4f2c10' sqlid='gvrhgcx5khbxd'
SELECT USER_ID, FIRST_NAME, LAST_NAME, USER_NAME, SECURE_CONTACT_INFO, ACTIVE, DISABLED, UPDATED_DATE_TIME FROM USERS WHERE "UPDATED_DATE_TIME" > :1  AND "UPDATED_DATE_TIME" < :2  ORDER BY "UPDATED_DATE_TIME" ASC
END OF STMT
PARSE #139624246405640:c=107,e=107,p=0,cr=0,cu=0,mis=0,r=0,dep=0,og=1,plh=1529657416,tim=14055447493172
BINDS #139624246405640:

 Bind#0
  oacdty=180 mxl=11(11) mxlc=00 mal=00 scl=00 pre=00
  oacflg=03 fl2=9000000 frm=00 csi=00 siz=32 off=0
  kxsbbbfp=7efccd60f140  bln=11  avl=07  flg=05
  value=31-AUG-23 08.35.09 AM
 Bind#1
  oacdty=180 mxl=11(11) mxlc=00 mal=00 scl=09 pre=00
  oacflg=03 fl2=9000000 frm=00 csi=00 siz=0 off=16
  kxsbbbfp=7efccd60f150  bln=11  avl=11  flg=01
  value=31-AUG-23 12.59.16.130000000 PM
EXEC #139624246405640:c=138,e=138,p=0,cr=0,cu=0,mis=0,r=0,dep=0,og=1,plh=1529657416,tim=14055447493354
WAIT #139624246405640: nam='SQL*Net message to client' ela= 1 driver id=1413697536 #bytes=1 p3=0 obj#=294182 tim=14055447493379

*** 2023-08-31T08:59:16.248853-04:00
FETCH #139624246405640:c=109735,e=117211,p=0,cr=30509,cu=0,mis=0,r=0,dep=0,og=1,plh=1529657416,tim=14055447610606
STAT #139624246405640 id=1 cnt=0 pid=0 pos=1 obj=0 op='SORT ORDER BY (cr=30509 pr=0 pw=0 str=1 time=117222 us cost=5238 size=76 card=1)'
STAT #139624246405640 id=2 cnt=0 pid=1 pos=1 obj=0 op='FILTER  (cr=30509 pr=0 pw=0 str=1 time=117200 us)'
STAT #139624246405640 id=3 cnt=0 pid=2 pos=1 obj=294182 op='TABLE ACCESS FULL USERS (cr=30509 pr=0 pw=0 str=1 time=117193 us cost=5237 size=76 card=1)'
WAIT #139624246405640: nam='SQL*Net message from client' ela= 387 driver id=1413697536 #bytes=1 p3=0 obj#=294182 tim=14055447611217
CLOSE #139624246405640:c=10,e=9,dep=0,type=1,tim=14055447611259

converted it to
-----------
set curn139624246405640 [oraopen $lda ]
set sql139624246405640 "SELECT USER_ID, FIRST_NAME, LAST_NAME, USER_NAME, SECURE_CONTACT_INFO, ACTIVE, DISABLED, UPDATED_DATE_TIME FROM USERS WHERE \"UPDATED_DATE_TIME\" > :1 AND \"UPDATED_DATE_TIME\" < :2 ORDER BY \"UPDATED_DATE_TIME\" ASC "
orasql $curn139624246405640 $sql139624246405640 -parseonly
#####CANNOT EXECUTE 139624246405640:Failed to find expected bind variables in statement before execute
###CANNOT FETCH 139624246405640:Failed to convert corresponding execute**
oraclose $curn139624246405640
oralogoff $lda

**Comment by sm-shaw** [comment]
This functionality can (and has been) impacted from time to time when Oracle change the trace file format which is not supported for external tools.  
There remains a lot of scope for enhancement depending on how much the feature is used, however it has not been updated much recently. 
Saying that, I have opened Issue #590 as this does look like a trace file format change issue which can be fixed.

**Reply by sm-shaw** [reply, marked as answer]
Pull Request #591 fixes the Issue
